iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractAreaTest.java

```diff
@@ -0,0 +1,201 @@
+
+        assertEquals("aic_a", aicA.getId());
+        assertEquals("Aic_A", aicA.getNameOrId());
+        assertEquals(aic, aicA.getAreaType());
+        assertEquals(Optional.of(10.0), aicA.getAcNetInterchangeTarget());
+        assertEquals(Optional.of(0.1), aicA.getAcNetInterchangeTolerance());
+
+        VoltageLevel vlhv2 = network.getVoltageLevel("VLHV2");
+        assertEquals(Set.of(vlhv2), aicA.getVoltageLevels());
+        assertEquals(List.of(vlhv2), aicA.getVoltageLevelStream().toList());
+
+        final Terminal boundary1 = network.getLine(EurostagTutorialExample1Factory.NHV1_NHV2_1).getTerminal1();
+        final Terminal boundary2 = network.getLine(EurostagTutorialExample1Factory.NHV1_NHV2_2).getTerminal2();
+        final Terminal boundary3 = network.getGenerator("GEN").getTerminal();
+        final List<Area.BoundaryTerminal> boundaries = List.of(new Area.BoundaryTerminal(boundary1, true),
+                new Area.BoundaryTerminal(boundary2, false), new Area.BoundaryTerminal(boundary3, true));
+        assertEquals(boundaries, aicA.getBoundaryTerminals());
+        assertEquals(boundaries, aicA.getBoundaryTerminalStream().toList());
+    }
+
+    @Test
+    void areaIterableAndStreamGetterCheck() {
+        List<Area> areas = List.of(biddingZoneA, biddingZoneB, regionA, aicA);
+        List<AreaType> areaTypes = List.of(biddingZone, region, aic);
+
+        assertEquals(areas, network.getAreaStream().toList());
+        assertEquals(areaTypes, network.getAreaTypeStream().toList());
+
+        assertEquals(4, Iterables.size(network.getAreas()));
+        areas.forEach(area -> assertTrue(Iterables.contains(network.getAreas(), area)));
+        assertEquals(3, Iterables.size(network.getAreaTypes()));
+        areaTypes.forEach(areaType -> assertTrue(Iterables.contains(network.getAreaTypes(), areaType)));
+    }
+
+    @Test
+    void checkSubnetworkGetAreas() {
+        Network subnetwork = network.createSubnetwork("subnetwork_id", "Subnetwork", "json");
+
+        VoltageLevel sn1VL1 = subnetwork.newVoltageLevel()
+                .setId("sub1_vl1")
+                .setNominalV(400.0)
+                .setTopologyKind(TopologyKind.NODE_BREAKER)
+                .add();
+
+        assertEquals(List.of(), subnetwork.getAreaTypeStream().toList());
+        assertEquals(List.of(), subnetwork.getAreaStream().toList());
+
+        biddingZoneA.addVoltageLevel(sn1VL1);
+
+        assertEquals(List.of(biddingZone), subnetwork.getAreaTypeStream().toList());
+        assertEquals(List.of(biddingZoneA), subnetwork.getAreaStream().toList());
+
+        assertEquals(1, Iterables.size(subnetwork.getAreas()));
+        assertTrue(Iterables.contains(subnetwork.getAreas(), biddingZoneA));
+        assertEquals(1, Iterables.size(subnetwork.getAreaTypes()));
+        assertTrue(Iterables.contains(subnetwork.getAreaTypes(), biddingZone));
+    }
+
+    @Test
+    void checkSubnetworkNewAreas() {
```

Comment:
Could you add a test with the following characteristics?
- the network has an area;
- this Area is not "visible" from the subnetwork;
- we try to create the "same" area from the subnetwork.

What is the expected behavior in this case? I think it should not fail, but the "new" area should be the same as the existing?